lib/openstack_query/docs/developer_docs/OVERVIEW.md

```diff
@@ -0,0 +1,92 @@
+# Developer Notes
+
+## Workflow Links
+- #### [Adding a new Query Feature](ADDING_NEW_QUERIES.md)
+- #### [Adding a new Preset](ADDING_NEW_PRESETS.md)
+- #### [Other smaller workflows](OTHER_WORKFLOWS.md)
+
+## Other Links
+- #### [Chaining Notes](CHAINING.md)
+
+
+## Overview
+
+The Query Library is broken up into 3 main parts.
+- Query Blocks
+  - which contain actual query logic split up into different parts - builder, executer, etc.
+- Mappings
+  - contains info that is specific to the query, used to setup query blocks
+- API
+  - exposes API methods to the user
+
+The image below shows how these components relate to each other when invoking a query object via API
+
+![Query API Workflow](./imgs/queryapi-workflow.png)
```

Comment:
Why is the `FlavorMapping` box highlighted in the image?

Reply:
oh that's just my bad
I'll upload another image